aws/config.go

```diff
@@ -43,6 +46,13 @@ type Config struct {
 	// client requests will be handled. This is useful for adding additional
 	// tracing data to a request, or changing behavior of the SDK's client.
 	APIOptions []func(*middleware.Stack) error
+
+	// The logger writer interface to write logging messages to. Defaults to
+	// standard error.
+	Logger logging.Logger
+
+	// Configures the events that will be sent to the configured logger.
+	ClientLogMode ClientLogMode
```

Comment:
would be good to callout documentation of these and how they will be used by the client potentially.